### DISCUSSION TITLE
[2024-05-27] Team GUI Demo Day

### DISCUSSION THREAD

**Discussion by kazcw** [enso-org/enso#10083] [Weekly Demo Day]
# Documentation editor
- #9935
- #9969
- #10051
- #10008

https://github.com/enso-org/enso/assets/1047859/fd40bee4-5a71-4b6a-a022-40b5c7ec8bec

# Component browser
- #9210

https://github.com/enso-org/enso/assets/1047859/a36a8a55-6717-4887-a72c-2b2eafde4260

# Bugs and misc
- #9936
- #9831
- #9903
- #9963
- #10057
- #10007

**Comment by Frizi** [comment]
- Added opensearch telemetry events to dashboard and GUI. That includes significant refactor of how the GUI application is embedded within the dashboard, bringing us closer to the goal of merging the apps together. https://github.com/enso-org/enso/pull/9951
- Implemented automatic documentation panel opening when function documentation is present. The doc panel state is stored in client's localStorage. https://github.com/enso-org/enso/pull/9990
- Attempted to use dashboard's react components for tooltips in GUI. Unfortunately it ended up not working due to its implementation details. We plan to reimplement them in Vue. Right now standard browser "title" is used instead.  https://github.com/enso-org/enso/issues/9828

**Comment by farmaazon** [comment]
* #9946 - small fix for visualization resizing when not zoom 100% + restored saving visualization sizes in project metadata.
* #9982
* #10001 - this test was failing randomly 
* #10002

![image](https://github.com/enso-org/enso/assets/3919101/40aa628c-4181-487c-a0f7-b49e06d274b8)

* #9959
* #10039
* #10047
* #10069 - fixing another set of random test failures

**Comment by JaroslavTulach** [comment]
Visible engine improvements:
- all is green, approvals collected: #9862
- `private` methods, more **private** of _everything_: #10060
- _reactive runtime cache_: #10065

[ReactiveCache.webm](https://github.com/enso-org/enso/assets/26887752/b00b7eb2-02cb-4042-8f23-e7f51a988a25)

Here is a demo of current functionality. The `b` argument of `as_list` function uses `cache` to read value of `a` and `b` arguments. The video shows the dropdown items are updated whenever the arguments change.
